[Kura 2.1.0 QA] Exceptions while deploying example service

While testing DEN-DEPLOY-001 the following exception hit the log file:

```
2016-12-07 15:51:23,529 [qtp23600506-28] INFO  o.e.k.w.s.s.FileServlet - Installing package...
2016-12-07 15:51:23,534 [DeploymentAgent] INFO  o.e.k.d.a.i.DeploymentAgent - About to install package at URL file:/tmp/org.eclipse.kura.example.publisher.dp
2016-12-07 15:51:24,768 [DeploymentAgent] INFO  o.e.k.c.c.ConfigurationServiceImpl - Registering metatype pid: org.eclipse.kura.example.publisher.ExamplePublisher with ocd: org.eclipse.kura.core.configuration.metatype.Tocd@78772c ...
2016-12-07 15:51:24,779 [DeploymentAgent] INFO  o.e.k.c.c.ConfigurationServiceImpl - Merging configuration for pid: org.eclipse.kura.example.publisher.ExamplePublisher
2016-12-07 15:51:24,800 [DeploymentAgent] ERROR o.e.k.c.c.ComponentMetaTypeBundleTracker - Error seeding configuration for pid: org.eclipse.kura.example.publisher.ExamplePublisher
org.eclipse.kura.KuraException: An internal error occurred. {0}
	at org.eclipse.kura.core.configuration.ConfigurationServiceImpl.registerComponentOCD(ConfigurationServiceImpl.java:506)
	at org.eclipse.kura.core.configuration.ComponentMetaTypeBundleTracker.processBundleMetaType(ComponentMetaTypeBundleTracker.java:96)
	at org.eclipse.kura.core.configuration.ComponentMetaTypeBundleTracker.addingBundle(ComponentMetaTypeBundleTracker.java:58)
	at org.eclipse.kura.core.configuration.ComponentMetaTypeBundleTracker.addingBundle(ComponentMetaTypeBundleTracker.java:1)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:467)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:1)
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
	at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:443)
	at org.eclipse.osgi.framework.internal.core.BundleContextImpl.dispatchEvent(BundleContextImpl.java:847)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.osgi.framework.internal.core.Framework.publishBundleEventPrivileged(Framework.java:1568)
	at org.eclipse.osgi.framework.internal.core.Framework.publishBundleEvent(Framework.java:1504)
	at org.eclipse.osgi.framework.internal.core.Framework.publishBundleEvent(Framework.java:1499)
	at org.eclipse.osgi.framework.internal.core.BundleHost.startWorker(BundleHost.java:391)
	at org.eclipse.osgi.framework.internal.core.AbstractBundle.start(AbstractBundle.java:300)
	at org.eclipse.osgi.framework.internal.core.AbstractBundle.start(AbstractBundle.java:292)
	at org.apache.felix.deploymentadmin.spi.StartBundleCommand.execute(StartBundleCommand.java:58)
	at org.apache.felix.deploymentadmin.spi.DeploymentSessionImpl.call(DeploymentSessionImpl.java:73)
	at org.apache.felix.deploymentadmin.DeploymentAdminImpl.installDeploymentPackage(DeploymentAdminImpl.java:214)
	at org.eclipse.kura.deployment.agent.impl.DeploymentAgent.installDeploymentPackageInternal(DeploymentAgent.java:469)
	at org.eclipse.kura.deployment.agent.impl.DeploymentAgent.installer(DeploymentAgent.java:309)
	at org.eclipse.kura.deployment.agent.impl.DeploymentAgent.access$0(DeploymentAgent.java:293)
	at org.eclipse.kura.deployment.agent.impl.DeploymentAgent$1.run(DeploymentAgent.java:167)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: No such file or directory
	at java.io.UnixFileSystem.createFileExclusively(Native Method)
	at java.io.File.createNewFile(File.java:1012)
	at org.eclipse.equinox.internal.cm.ConfigurationStore.writeConfigurationFile(ConfigurationStore.java:112)
	at org.eclipse.equinox.internal.cm.ConfigurationStore$1.run(ConfigurationStore.java:99)
	at java.security.AccessController.doPrivileged(Native Method)
	at org.eclipse.equinox.internal.cm.ConfigurationStore.saveConfiguration(ConfigurationStore.java:97)
	at org.eclipse.equinox.internal.cm.ConfigurationImpl.update(ConfigurationImpl.java:255)
	at org.eclipse.kura.core.configuration.ConfigurationServiceImpl.updateWithDefaultConfiguration(ConfigurationServiceImpl.java:691)
	at org.eclipse.kura.core.configuration.ConfigurationServiceImpl.registerComponentOCD(ConfigurationServiceImpl.java:504)
	... 29 more
2016-12-07 15:51:24,882 [DeploymentAgent] INFO  o.e.k.d.a.i.DeploymentAgent - Posting INSTALLED event for package at URL file:/tmp/org.eclipse.kura.example.publisher.dp: successful

```

--- a/kura/distrib/src/main/ant/build_equinox_distrib.xml
+++ b/kura/distrib/src/main/ant/build_equinox_distrib.xml
@@ -354,6 +354,8 @@ fi]]>
                         prefix="${build.output.name}/install" />
 			<zipfileset file="src/main/resources/common/monitrc.raspbian"
                         prefix="${build.output.name}/install" />
+            <zipfileset file="src/main/resources/${build.name}/kura-tmpfiles.conf"
+                        prefix="${build.output.name}/install" />
 
 
 			<zipfileset file="src/main/resources/common/logrotate.conf"

--- a/kura/distrib/src/main/resources/fedorapi/kura-tmpfiles.conf
+++ b/kura/distrib/src/main/resources/fedorapi/kura-tmpfiles.conf
@@ -1,0 +1,3 @@
+# Kura specific file. Do not edit!
+
+x /tmp/.kura

--- a/kura/distrib/src/main/resources/fedorapi/kura_install.sh
+++ b/kura/distrib/src/main/resources/fedorapi/kura_install.sh
@@ -99,3 +99,6 @@ if [ ! -d /etc/logrotate.d/ ]; then
     mkdir -p /etc/logrotate.d/
 fi
 cp ${INSTALL_DIR}/kura/install/kura.logrotate /etc/logrotate.d/kura
+
+# Setup tmpfiles.d
+cp ${INSTALL_DIR}/kura/install/kura-tmpfiles.conf /usr/lib/tmpfiles.d/kura.conf

